### ISSUE TITLE
qitmeer fails on start after cleaned

### ISSUE THREAD

**Issue by forchain** [Qitmeer/qitmeer#160]
$ ./qitmeer -L
$ ./qitmeer
2019-11-01|04:46:15.352 [ERROR] load block database                 error="write cursor does not exist"
2019-11-01|04:46:15.352 [INFO ] Shutdown complete

**Comment by forchain** [comment]
launching succeeded if ~/.qitmeerd/data/testnet/blocks_ffldb deleted

**Comment by dindinw** [comment]
I can't reproduce the issue 
```
$ ./qitmeer -b . --testnet -L
2019-11-01|11:06:40.167 [INFO ] System info                         Qitmeer Version=0.7.7+dev Go version=go1.13
2019-11-01|11:06:40.167 [INFO ] Loading block database              dbPath=testnet/blocks_ffldb
2019-11-01|11:06:40.197 [INFO ] Block database loaded
2019-11-01|11:06:40.197 [INFO ] Removing block database from 'testnet/blocks_ffldb'
2019-11-01|11:06:40.201 [INFO ] Finished cleanup
2019-11-01|11:06:40.201 [INFO ] Gracefully shutting down the database...
2019-11-01|11:06:40.202 [INFO ] Shutdown complete
```
```
$ ./start.sh
2019-11-01|11:07:04.373 [INFO |qitmeer/qitmeerd.go:63]       System info                        Qitmeer Version=0.7.7+dev Go version=go1.13
2019-11-01|11:07:04.373 [INFO |/services/common/db.go:29]    Loading block database             dbPath=/data/qitmeer/testnet/blocks_ffldb
2019-11-01|11:07:04.404 [INFO |/services/common/db.go:49]    Block database loaded
2019-11-01|11:07:04.405 [INFO |/node/node.go:126]            Starting Server                    module=node
2019-11-01|11:07:04.406 [INFO |/node/full.go:113]            Transaction index is enabled       module=node
2019-11-01|11:07:04.406 [INFO |/core/blockdag/phantom.go:44] anticone size:3                    module=blockdag
2019-11-01|11:07:04.406 [INFO |/services/index/indexmgr.go:192] Catching up indexes from order -1 to 0
2019-11-01|11:07:04.407 [INFO |/services/index/indexmgr.go:248] Indexes caught up to order 0
2019-11-01|11:07:04.407 [INFO |/core/blockchain/blockchain.go:288] DAG Type:phantom                   module=blockchain
2019-11-01|11:07:04.407 [INFO |/core/blockchain/blockchain.go:289] Blockchain database version        module=blockchain chain=1 compression=1 index=1
2019-11-01|11:07:04.407 [INFO |/core/blockchain/blockchain.go:293] Chain state:totaltx=1 tipsNum=1 mainOrder=0 total=1module=blockchain
2019-11-01|11:07:04.407 [INFO |/core/blockchain/blockchain.go:297] hash=01fec9d4be0fe93b27261df1305b119573c5830002df5687aa05a4b79dd39615,order=0,work=34359754752module=blockchain
2019-11-01|11:07:04.408 [INFO |/services/tx/txmanager.go:40]       Starting tx manager                module=txmanager
2019-11-01|11:07:04.409 [INFO |/p2p/addmgr/addrmgr.go:426]         Loaded 4 addresses from file '/data/qitmeer/testnet/peers.json'
2019-11-01|11:07:04.409 [INFO |/p2p/connmgr/connmgr.go:60]         Connection manager started
2019-11-01|11:07:04.409 [INFO |/p2p/connmgr/connmgr.go:557]        P2P Server listening on            addr=[::]:18130
2019-11-01|11:07:04.410 [INFO |/p2p/connmgr/connmgr.go:557]        P2P Server listening on            addr=0.0.0.0:18130
2019-11-01|11:07:04.420 [INFO |/rpc/rpcserver.go:192]              RPC server listening on            addr=127.0.0.1:1234
2019-11-01|11:07:04.672 [INFO |/services/blkmgr/handle_peer.go:28] New valid peer: 47.88.220.44:18130 (outbound),user-agent:qitmeer:0.7.6module=blkmanager
2019-11-01|11:07:04.672 [INFO |/services/blkmgr/handle_peer.go:151] Syncing to state (7535,7493,7493,7536,1) from peer 47.88.220.44:18130 cur graph state:(0,0,0,1,1)module=blkmanager
2019-11-01|11:07:04.672 [INFO |/services/blkmgr/handle_peer.go:28]  New valid peer: 47.74.183.189:18130 (outbound),user-agent:qitmeer:0.7.6module=blkmanager
2019-11-01|11:07:04.753 [INFO |/services/blkmgr/handle_peer.go:28]  New valid peer: 103.15.132.180:18130 (outbound),user-agent:qitmeer:0.7.6module=blkmanager
2019-11-01|11:07:05.187 [INFO |/services/blkmgr/handle_peer.go:28]  New valid peer: 47.103.194.115:18130 (outbound),user-agent:qitmeer:0.7.6module=blkmanager
2019-11-01|11:07:05.418 [WARN |/p2p/connmgr/seed.go:50]             DNS discovery failed               seed=testnet-seed.qitmeer.xyz error="lookup testnet-seed.qitmeer.xyz: no such host"
2019-11-01|11:07:05.418 [INFO |/p2p/connmgr/seed.go:55]             4 addresses found from DNS seed testnet-seed.hlcwallet.info
2019-11-01|11:07:07.420 [WARN |/p2p/connmgr/seed.go:50]             DNS discovery failed               seed=testnet-seed.qitmeer.top error="lookup testnet-seed.qitmeer.top: no such host"
2019-11-01|11:07:15.018 [INFO |/services/common/progresslog/blocklogger.go:113] Processed 1856 blocks in the last 10.61s (1864 transactions, order 1856, 2019-10-30 11:21:20 +0800 CST)module=blkmanager
2019-11-01|11:07:25.018 [INFO |/services/common/progresslog/blocklogger.go:113] Processed 2114 blocks in the last 10s (2449 transactions, order 3970, 2019-10-31 04:38:17 +0800 CST)module=blkmanager
2019-11-01|11:07:35.493 [INFO |/services/common/progresslog/blocklogger.go:113] Processed 2434 blocks in the last 10.47s (2441 transactions, order 6404, 2019-11-01 01:05:59 +0800 CST)module=blkmanager
2019-11-01|11:08:31.567 [INFO |/services/common/progresslog/blocklogger.go:113] Processed 1124 blocks in the last 56.07s (1128 transactions, order 7538, 2019-11-01 11:07:43 +0800 CST)module=blkmanager
2019-11-01|11:08:38.907 [INFO |/services/blkmgr/handle_block.go:172]            Your synchronization has been completed. module=blkmanager
```

**Comment by forchain** [comment]
reproduced in Windows


Pro@Miner ~/github.com/Qitmeer/qitmeer (master*) $ ./qitmeer -L
2019-11-01|16:37:58.653 [INFO ] System info                         Qitmeer Version=0.7.7+dev Go version=go1.13.1
2019-11-01|16:37:58.654 [INFO ] System info                         UUID=d5c176a1-4150-4c62-901b-38249894dd3a
2019-11-01|16:37:58.654 [INFO ] System info                         Home dir=C:\\Users\\Pro\\AppData\\Local\\Qitmeerd
2019-11-01|16:37:58.654 [INFO ] Loading block database              dbPath=C:\\Users\\Pro\\AppData\\Local\\Qitmeerd\\data\\testnet\\blocks_ffldb
2019-11-01|16:37:58.672 [INFO ] Detected unclean shutdown - Repairing... module=database
2019-11-01|16:37:58.673 [INFO ] Database sync complete              module=database
2019-11-01|16:37:58.673 [INFO ] Block database loaded
2019-11-01|16:37:58.673 [INFO ] Removing block database from 'C:\Users\Pro\AppData\Local\Qitmeerd\data\testnet\blocks_ffldb'
2019-11-01|16:37:58.717 [INFO ] Finished cleanup
2019-11-01|16:37:58.717 [INFO ] Gracefully shutting down the database...
2019-11-01|16:37:58.725 [INFO ] Shutdown complete
Pro@Miner ~/github.com/Qitmeer/qitmeer (master*) $ ./qitmeer.exe
2019-11-01|16:38:02.821 [INFO ] System info                         Qitmeer Version=0.7.7+dev Go version=go1.13.1
2019-11-01|16:38:02.821 [INFO ] System info                         UUID=86cebe62-c621-4349-a54d-b9974cd76dfd
2019-11-01|16:38:02.821 [INFO ] System info                         Home dir=C:\\Users\\Pro\\AppData\\Local\\Qitmeerd
2019-11-01|16:38:02.821 [INFO ] Loading block database              dbPath=C:\\Users\\Pro\\AppData\\Local\\Qitmeerd\\data\\testnet\\blocks_ffldb
2019-11-01|16:38:02.827 [ERROR] load block database                 error="file missing [file=MANIFEST-000000]: file missing [file=MANIFEST-000000]"
2019-11-01|16:38:02.827 [INFO ] Shutdown complete

**Comment by forchain** [comment]
Probably it is related to the platform,
now I am only able to reproduce on Windows

**Comment by lochjin** [comment]
> $ ./qitmeer -L
> $ ./qitmeer
> 2019-11-01|04:46:15.352 [ERROR] load block database error="write cursor does not exist"
> 2019-11-01|04:46:15.352 [INFO ] Shutdown complete

Solved：https://github.com/Qitmeer/qitmeer/pull/161

**Comment by dindinw** [comment]
fixed by #161
